### DISCUSSION TITLE
Create new workspace and insert it between two existing ones

### DISCUSSION THREAD

**Discussion by metov** [i3/i3#4761] [Q&A] [answered]
Is there an i3 command already for inserting a new workspace? For example, say I have WSes `a b c`. I am on `a`. I use this command, and create `d`. The result would be `a d b c`.

I believe that I can do this by simply writing a script that uses `i3-msg` to shuffle the WS around. However it seems like a pretty basic function, so I wonder if it's already implemented.

**Comment by Airblader** [comment, marked as answer]
No, there is no command to explicitly reorder workspaces in i3bar. You can either do what you said or work with workspace numbers (i3bar sorts workspaces by their number), possibly hiding them using `strip_workspace_numbers` if you don't want them to be shown.
